python/lsst/pipe/supertask/parser.py

```diff
@@ -0,0 +1,278 @@
+# overrides can be applied later.
+_Override = collections.namedtuple("_Override", "type,value")
+
+
+class _LogLevelAction(Action):
+    """Action class which collects logging levels.
+    """
+
+    permittedLevels = set(['TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'])
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        dest = getattr(namespace, self.dest)
+        if dest is None:
+            dest = []
+            setattr(namespace, self.dest, dest)
+        for componentLevel in values:
+            component, sep, levelStr = componentLevel.partition("=")
+            if not levelStr:
+                levelStr, component = component, None
+            logLevelUpr = levelStr.upper()
+            if logLevelUpr not in self.permittedLevels:
+                parser.error("loglevel=%s not one of %s" % (levelStr, tuple(self.permittedLevels)))
+            dest.append((component, logLevelUpr))
+
+
+class _AppendFlattenAction(Action):
+    """Action class which appends all items of multi-value option to the
+    same destination.
+
+    This is different from standard 'append' action which does not flatten
+    the option values.
+    """
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        dest = getattr(namespace, self.dest)
+        if dest is None:
+            dest = []
+            setattr(namespace, self.dest, dest)
+        dest += values
+
+
+def _config_override(value):
+    """This callable is used as "type" for -c option.
+    """
+    return _Override("override", value)
+
+
+def _config_file(value):
+    """This callable is used as "type" for -C option.
+    """
+    return _Override("file", value)
+
+
+_epilog = """\
+Notes:
+  * --config, --configfile, --id, --loglevel and @file may appear multiple times;
+    all values are used, in order left to right
+  * @file reads command-line options from the specified file:
+    * data may be distributed among multiple lines (e.g. one option per line)
+    * data after # is treated as a comment and ignored
+    * blank lines and lines starting with # are ignored
+  * To specify multiple values for an option, do not use = after the option name:
+    * right: --configfile foo bar
+    * wrong: --configfile=foo bar
+"""
+
+#------------------------
+# Exported definitions --
+#------------------------
+DEFAULT_INPUT_NAME = base_parser.DEFAULT_INPUT_NAME
+DEFAULT_CALIB_NAME = base_parser.DEFAULT_CALIB_NAME
+DEFAULT_OUTPUT_NAME = base_parser.DEFAULT_OUTPUT_NAME
+
+
+def makeParser(fromfile_prefix_chars='@', parser_class=ArgumentParser, **kwargs):
+    """Class implementing command line parsing for
+    :py:`lsst.pipe.supertask.CmdLineActivator`.
+
+    This class inherits from standard argparse.ArgumentParser and adds all options
+    that are supported by command line activator. There is no additional logic in
+    this class, all semantics is handled by the activator class.
+
```

Comment:
Yes, my fault, there was a class at the beginning but then I decided to make a method instead and never updated its comment. Will fix it.